aten/src/ATen/native/Convolution.cpp

```diff
@@ -119,6 +123,28 @@ auto ConvParams::view1d_as_2d() -> void {
   }
 }
 
+bool ConvParams::use_cpu_depthwise3x3_winograd(
+    const at::Tensor& input, const at::Tensor& weight) const
+{
+#ifdef __ARM_NEON__
```

Comment:
OK so it seems that source files under the cpu directory are compiled multiple times, each time with compiler flags set slightly differently to optimize code generation for different CPU instruction sets (mainly x86 AVX and AVX2 as it stands if I am reading the code correctly) which means any symbol that has external linkage will end up colliding with other definitions of itself as a result of multiples of such definitions getting introduced each per target compiled.  Also as an aside, this is the reason behind the emphasis on wrapping symbols inside anonymous namespaces there to force internal linkage.  That means that unless I am mistaken, exposing a symbol from those files is not possible, which is why I had to pull this #ifdef out and into this file. :/

Reply:
ah, here's an idea - for REGISTER_DISPATCH in another file put either the function or `static_cast<fn_type>(nullptr)` depending the #ifdef too. Then here you can check `convolution_depthwise3x3_winograd_stub::DEFAULT` at runtime to see whether it's populated. To you point of compiling several times - REGISTER_DISPATCH would resolve to different variables in this case

but it's up to you - the current approach is also ok